File: src/main/java/com/hsmchurch/app/video/VideoConfiguration.java
package com.hsmchurch.app.video;

import com.hsmchurch.app.video.support.DescriptionParser;
import com.hsmchurch.app.video.support.YoutubeDescriptionParser;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;

@Configuration
public class VideoConfiguration {

    @Bean(name = "descriptionParser")
    public DescriptionParser descriptionParser() {
        return new YoutubeDescriptionParser();
    }

}

File: src/main/java/com/hsmchurch/app/video/VideoConfiguration.kt
package com.hsmchurch.app.video

import com.hsmchurch.app.video.support.DescriptionParser
import com.hsmchurch.app.video.support.YoutubeDescriptionParser
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration

@Configuration
open class VideoConfiguration {

    @Bean(name = ["descriptionParser"])
    open fun descriptionParser(): DescriptionParser = YoutubeDescriptionParser()

}
